lib/linked_list.rb

```diff
@@ -10,36 +10,77 @@ def initialize
   # Time complexity - ?
   # Space complexity - ?
   def add_first(data)
-
+    new_node = Node.new(data)
+    new_node.next = @head
+    @head = new_node
   end
 
   # Time complexity - ?
   # Space complexity - ?
   def get_first
-
+    if @head != nil
+      return @head.data
+    else
+      return nil
+    end
   end
 
   # Time complexity - ?
   # Space complexity - ?
   def length
-    return 0
+    current = @head
+    count = 0
+
+    while current != nil
+      count +=1
+      current = current.next
+    end
+    return count
   end
 
   # Time complexity - ?
   # Space complexity - ?
   def add_last(data)
-
+    new_node = Node.new(data)
+    
+    if @head == nil
+      add_first(data)
+    else  
+      current = @head
+      while current.next != nil
+        current = current.next
+      end
+      current.next = new_node
+    end
   end
 
   # Time complexity - ?
   # Space complexity - ?
   def get_last
+    current = @head
```

Comment:
What if `@head` is `nil`?

```suggestion
    return nil if @head.nil?
    current = @head
```